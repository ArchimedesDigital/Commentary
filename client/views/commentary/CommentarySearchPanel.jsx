import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import TextField from 'material-ui/TextField';
import Drawer from 'material-ui/Drawer';
import { Card, CardHeader, CardText } from 'material-ui/Card';

CommentarySearchPanel = React.createClass({

	propTypes: {
		filters: React.PropTypes.array,
		toggleSearchTerm: React.PropTypes.func,
		handleChangeTextsearch: React.PropTypes.func,
		handleChangeLineN: React.PropTypes.func,
		open: React.PropTypes.bool,
		closeRightMenu: React.PropTypes.func,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	mixins: [ReactMeteorData],

	getInitialState() {
		return {
			subworks: [],
			activeWork: '',
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	getMeteorData() {
		return {
			keywords: Keywords.find({
				type: 'word',
				count: { $gt: 0 },
			}, { sort: { title: 1 } }).fetch(),
			keyideas: Keywords.find({
				type: 'idea',
				count: { $gt: 0 },
			}, { sort: { title: 1 } }).fetch(),
			commenters: Commenters.find({}, { sort: { name: 1 } }).fetch(),
			works: Works.find({}, { sort: { order: 1 } }).fetch(),
			subworks: Subworks.find({}, { sort: { n: 1 } }).fetch(),
		};
	},

	toggleSearchTerm(key, value) {
		this.props.toggleSearchTerm(key, value);
	},

	toggleWorkSearchTerm(key, value) {
		const work = value;
		const newValue = value;
		newValue.subworks.forEach((subwork, i) => {
			newValue.subworks[i].work = work;
		});

		if (this.state.activeWork === newValue.slug) {
			this.setState({
				subworks: [],
				activeWork: '',
			});
		} else {
			newValue.subworks.sort((a, b) => {
				if (a.n < b.n) {
					return -1;
				}
				if (a.n > b.n) {
					return 1;
				}
				return 0;
			});
			this.setState({
				subworks: newValue.subworks,
				activeWork: newValue.slug,
			});
		}

		this.props.toggleSearchTerm(key, newValue);
	},

	handleChangeTextsearch(event) {
		this.props.handleChangeTextsearch(event.target.value);
	},

	render() {
		const self = this;
		const filters = this.props.filters || [];

		const styles = {
			flatButton: {
				width: 'auto',
				minWidth: 'none',
				height: '80px',
				padding: '21px 5px',
			},
			flatIconButton: {
				padding: '10px 20px',
				width: 'auto',
				minWidth: 'none',
				height: '55px',
			},
			wrapper: {
				display: 'flex',
				flexWrap: 'wrap',
			},
			textSearch: {
				width: '100%',
				padding: '0px 10px',
				background: '#f2f2f2',
			},
			cardHeader: {
				fontFamily: 'Proxima N W01 At Smbd',
				textTransform: 'uppercase',
			},
			lineSearch: {
				width: '99%',
				margin: '0px auto',
			},
		};

		return (
			<Drawer
				openSecondary
				open={this.props.open}
				docked={false}
				onRequestChange={this.props.closeRightMenu}
			>
				<div style={styles.textSearch} className="search-tool text-search">
					<TextField
						hintText=""
						floatingLabelText="Search"
						fullWidth
						onChange={this.handleChangeTextsearch}
					/>
				</div>
				<Card>
					<CardHeader
						title="Line Number"
						style={styles.cardHeader}
						actAsExpander
						showExpandableButton
					/>
					<CardText expandable style={styles.wrapper}>
						<div style={styles.lineSearch} className="line-search">
							<LineRangeSlider
								handleChangeLineN={this.props.handleChangeLineN}
							/>
						</div>
					</CardText>
				</Card>
				<Card>
					<CardHeader
						title="Keywords"
						style={styles.cardHeader}
						actAsExpander
						showExpandableButton
					/>
					<CardText expandable style={styles.wrapper}>
						{self.data.keywords.map((keyword, i) => {
							let active = false;
							filters.forEach((filter) => {
								if (filter.key === 'keywords') {
									filter.values.forEach((value) => {
										if (keyword._id === value._id) {
											active = true;
										}
									});
								}
							});

							return (
								<SearchTermButtonPanel
									key={i}
									toggleSearchTerm={self.toggleSearchTerm}
									label={keyword.title}
									searchTermKey="keywords"
									value={keyword}
									active={active}
								/>
							);
						})}
						{self.data.keywords.length === 0 ?
							<div className="no-results">No keywords found in objects.</div>
							: ''
						}
					</CardText>
				</Card>
				<Card>
					<CardHeader
						title="Keyideas"
						style={styles.cardHeader}
						actAsExpander
						showExpandableButton
					/>
					<CardText expandable style={styles.wrapper}>
						{self.data.keyideas.map((keyidea, i) => {
							let active = false;
							filters.forEach((filter) => {
								if (filter.key === 'keyideas') {
									filter.values.forEach((value) => {
										if (keyidea._id === value._id) {
											active = true;
										}
									});
								}
							});

							return (
								<SearchTermButtonPanel
									key={i}
									toggleSearchTerm={self.toggleSearchTerm}
									label={keyidea.title}
									searchTermKey="keyideas"
									value={keyidea}
									active={active}
								/>
							);
						})}
						{self.data.keyideas.length === 0 ?
							<div className="no-results">No keyideas found in objects.</div>
							: ''
						}
					</CardText>
				</Card>
				<Card>
					<CardHeader
						title="Commenter"
						style={styles.cardHeader}
						actAsExpander
						showExpandableButton
					/>
					<CardText expandable style={styles.wrapper}>
						{self.data.commenters.map((commenter, i) => {
							let active = false;
							filters.forEach((filter) => {
								if (filter.key === 'commenters') {
									filter.values.forEach((value) => {
										if (commenter._id === value._id) {
											active = true;
										}
									});
								}
							});

							return (
								<SearchTermButtonPanel
									key={i}
									toggleSearchTerm={self.toggleSearchTerm}
									label={commenter.name}
									searchTermKey="commenters"
									value={commenter}
									active={active}
								/>
							);
						})}
						{self.data.commenters.length === 0 ?
							<div className="no-results">No commenters found in objects.</div>
							: ''
						}
					</CardText>
				</Card>
				<Card>
					<CardHeader
						title="Work"
						style={styles.cardHeader}
						actAsExpander
						showExpandableButton
					/>
					<CardText expandable style={styles.wrapper}>
						{self.data.works.map((work, i) => (
							<SearchTermButtonPanel
								key={i}
								toggleSearchTerm={self.toggleWorkSearchTerm}
								label={work.title}
								searchTermKey="works"
								value={work}
								activeWork={self.state.activeWork === work.slug}
							/>
						))}
						{self.data.works.length === 0 ?
							<div className="no-results">No works found in objects.</div>
							: ''
						}
					</CardText>
				</Card>
				<Card>
					<CardHeader
						title="Book"
						style={styles.cardHeader}
						actAsExpander={!(self.state.subworks.length === 0)}
						showExpandableButton={!(self.state.subworks.length === 0)}
					/>
					<CardText expandable style={styles.wrapper}>
						{self.state.subworks.map((subwork, i) => {
							let active = false;
							filters.forEach((filter) => {
								if (filter.key === 'subworks') {
									filter.values.forEach((value) => {
										if (subwork.n === value.n) {
											active = true;
										}
									});
								}
							});

							return (
								<SearchTermButtonPanel
									key={i}
									toggleSearchTerm={self.toggleSearchTerm}
									label={`${subwork.work.title} ${subwork.title}`}
									searchTermKey="subworks"
									value={subwork}
									active={active}
								/>
							);
						})}
						{self.data.subworks.length === 0 ?
							<div className="no-results">No subworks found in objects.</div>
							: ''
						}
					</CardText>
				</Card>
			</Drawer>
		);
	},
});
