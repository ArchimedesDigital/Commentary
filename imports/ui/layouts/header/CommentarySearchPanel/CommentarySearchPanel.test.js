import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import CommentarySearchPanel from './CommentarySearchPanel';

describe('CommentarySearchPanel', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<CommentarySearchPanel
						handleChangeLineN={() => {}}
						isTest
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
