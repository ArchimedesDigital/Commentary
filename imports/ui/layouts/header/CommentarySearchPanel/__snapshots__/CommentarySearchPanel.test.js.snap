// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentarySearchPanel renders correctly 1`] = `
<div
  className="search-tools-drawer"
  style={
    Object {
      "backgroundColor": "#ddd",
    }
  }
>
  <div
    className={undefined}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0.54)",
        "height": "100%",
        "left": "-100%",
        "muiPrepared": true,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "transform": "translateZ(0)",
        "transition": "left 0ms cubic-bezier(0.23, 1, 0.32, 1) 400ms, opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "willChange": "opacity",
        "zIndex": 1200,
      }
    }
  />
  <div
    className={undefined}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "0px",
        "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
               0 3px 10px rgba(0, 0, 0, 0.23)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "height": "100%",
        "left": "auto",
        "muiPrepared": true,
        "overflow": "auto",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "transform": "translate(410px, 0)",
        "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 400,
        "zIndex": 1300,
      }
    }
  >
    <div
      className="search-tool text-search text-search--drawer"
    >
      
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header"
          onTouchTap={[Function]}
          style={
            Object {
              "boxSizing": "border-box",
              "cursor": "pointer",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Work
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "bottom": 0,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": "auto",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header "
          style={
            Object {
              "boxSizing": "border-box",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Book
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header "
          style={
            Object {
              "boxSizing": "border-box",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Line Number
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header"
          onTouchTap={[Function]}
          style={
            Object {
              "boxSizing": "border-box",
              "cursor": "pointer",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Keywords
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "bottom": 0,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": "auto",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header"
          onTouchTap={[Function]}
          style={
            Object {
              "boxSizing": "border-box",
              "cursor": "pointer",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Key Ideas
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "bottom": 0,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": "auto",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header"
          onTouchTap={[Function]}
          style={
            Object {
              "boxSizing": "border-box",
              "cursor": "pointer",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Commentator
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "bottom": 0,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": "auto",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="search-tool-card"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="card-header"
          onTouchTap={[Function]}
          style={
            Object {
              "boxSizing": "border-box",
              "cursor": "pointer",
              "fontFamily": "Proxima N W01 At Smbd",
              "fontSize": "14px",
              "fontWeight": "bold",
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "paddingRight": "90px",
                "verticalAlign": "top",
                "whiteSpace": "normal",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "block",
                  "fontSize": 15,
                  "muiPrepared": true,
                }
              }
            >
              Reference
            </span>
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "block",
                  "fontSize": 14,
                  "muiPrepared": true,
                }
              }
            />
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "bottom": 0,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": "auto",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
